
# HG changeset patch
# User Kevin Daudt <kdaudt@alpinelinux.org>
# Date 1639316328 0
# Node ID 2ed1b3cb869bc6191bf306bab467ea42c3908c03
# Parent  d6b06f985900e549cf9b3e37562e25c2ef5c82f0
Bug 1745560 - Add missing stub for wl_proxy_marshal_flags, r=stransky, a=RyanVM

Firefox fails to build against wayland 1.20 because this symbol is missing

Differential Revision: https://phabricator.services.mozilla.com/D133583

diff --git a/widget/gtk/mozwayland/mozwayland.c b/widget/gtk/mozwayland/mozwayland.c
--- a/widget/gtk/mozwayland/mozwayland.c
+++ b/widget/gtk/mozwayland/mozwayland.c
@@ -195,8 +195,15 @@ MOZ_EXPORT void wl_list_insert(struct wl
 MOZ_EXPORT void wl_list_remove(struct wl_list* elm) {}
 
 MOZ_EXPORT int wl_list_length(const struct wl_list* list) { return -1; }
 
 MOZ_EXPORT int wl_list_empty(const struct wl_list* list) { return -1; }
 
 MOZ_EXPORT void wl_list_insert_list(struct wl_list* list,
                                     struct wl_list* other) {}
+
+MOZ_EXPORT struct wl_proxy* wl_proxy_marshal_flags(
+    struct wl_proxy* proxy, uint32_t opcode,
+    const struct wl_interface* interface, uint32_t version, uint32_t flags,
+    ...) {
+  return NULL;
+}

